src/app/components/api/overlayoptions.ts

```diff
@@ -0,0 +1,43 @@
+import { AnimationEvent } from '@angular/animations';
+
+export type OverlayModeType = 'modal' | 'overlay' | undefined;
+
+export type ResponsiveOverlayDirectionType = 'start' | 'center' | 'end' | undefined;
+
+export interface OverlayListenerOptions {
+    type?: 'scroll' | 'outside' | 'resize' | undefined;
+    mode?: OverlayModeType;
+    valid?: boolean;
+}
+
+export interface ResponsiveOverlayOptions {
+    style?: any;
+    styleClass?: string;
+    breakpoint?: string;
+    media?: string;
+    direction?: ResponsiveOverlayDirectionType;
+}
+
+export interface OverlayOnShowEvent {
+    container?: HTMLElement | undefined;
+    target?: HTMLElement | undefined;
+    mode?: OverlayModeType;
+}
+
+export interface OverlayOnHideEvent extends OverlayOnShowEvent {}
+
+export interface OverlayOptions {
+    style?: any;
+    styleClass?: string;
+    appendTo?: 'body' | HTMLElement | undefined;
```

Comment:
Hi, I think you should also add 'self' for type

```ts
 appendTo?: 'body' | 'self' | HTMLElement;
```

Reply:
Yes, this enhancement already exists in our roadmap. We will port a structure like in PrimeReact. https://www.primefaces.org/primereact/setup/